### ISSUE TITLE
Event Metadata fields are inconsistent 

### ISSUE THREAD

**Issue by Sachin4403** [xmidt-org/talaria#186]
Hi Team,

I was unmarshalling the  Node change and device status events, I am getting an unmarshalling error on partner_id in device status events as it is missing. 
We are getting this metadata in case of 

# Device status events
```
"/hw-serial-number"
"/hw-last-reboot-reason"
"/fw-name"
"/trust"
"/compliance"
"/boot-time"
"/hw-model"
"/hw-manufacturer"
```
# Node change 
```
"hw-model"
"boot-time"
"webpa-uuid"
"webpa-interface-used"
"partner-id"
"hw-serial-number"
"fw-name"
"hw-manufacturer"
"hw-mac"
"hw-last-reboot-reason"
"webpa-last-reconnect-reason"
"webpa-protocol"
```

 Can we make a single format for metadata in all the events?

cc @joe94 @schmidtw 

Note: Device status events are generated from talaria and node change events are generated from parodus2ccsp

**Comment by johnabass** [comment]
These nodes are set by the RDK software stack, not by xmidt/webpa.  The metadata sent to talaria in WRP messages is intentionally not controlled by talaria itself.

**Comment by Sachin4403** [comment]
Hi @johnabass  Can we update the metadata creation in case of CPE online/offline in xmidt/webpa?

As online/offline events will be created by xmidt/talaria.

**Comment by Sachin4403** [comment]
Hi @johnabass,

We will be getting these all fields in [convey header](https://github.com/xmidt-org/parodus/blob/dee98922a667a5725bef240ead6ee8d5a7c6261d/src/ParodusInternal.c#L36-L88) based on this  we can construct all the fields except `webpa-uuid`

let me know if this needs to be done from my end.

**Comment by Sachin4403** [comment]
Hi @johnabass 
One more bug is identified
we are setting `compliance=full` always if there are missing fileds in convey header.

The above logic will not able to find  [these 2 keys ](https://github.com/xmidt-org/talaria/blob/f104d3f4e39dfdfa8b1f0b44ef8590348e1ef495/deviceStatus.go#L23-L24)in convey header and will add `compliance=full`

https://github.com/xmidt-org/wrp-go/blob/b6f9bd049f12765ee89aad8e47dd35e5d0806905/wrpmeta/metadata.go#L93-L97

ideally, it should be 
```
			if present {
				b.product[to] = value
			} else{
				 if len(f.Default) > 0 {
					b.product[to] = f.Default
				}
				b.allFieldsPresent = false
			}
```

cc: @schmidtw

**Comment by johnabass** [comment]
Very sorry, but I'm still not sure I understand.

For clarification, neither `talaria` nor any software in the `xmidt` cluster controls the metadata fields.  Those are sent by the devices that connect to our cluster.  Are you asking for devices to send new fields?

`wrp-go` does some processing so that the fields show up as valid JSON.  The code you linked above does that, essentially stripping the leading "/".  Are you asking for a new field to be similarly processed?

**Comment by Sachin4403** [comment]
Hi @johnabass 
Sorry for replying late.
 > For clarification, neither talaria nor any software in the xmidt cluster controls the metadata fields. Those are sent by the devices that connect to our cluster. Are you asking for devices to send new fields?

I didn't mean this,  As a context of this issue, we are getting different fields  in online/offline event(originated from Talaria) and Node change event(originated from CPE)

now we want to make the fields consistent in all the events(originated from Talaria and originated from CPE).

Note: originated from Talaria Means then events which are created at the talaria and later sent to caduceus for further distribution. here is the source code for the same. https://github.com/xmidt-org/talaria/blob/main/deviceStatus.go#L14-L34. we are reading all the fields from convey token.

**Comment by denopink** [comment]
Closing since patched by xmidt-org/wrp-go#79
